environment.go

```diff
@@ -348,21 +353,12 @@ func (e *envState) initDB(ctx context.Context, settings DBSettings) error {
 		SetConnectTimeout(5 * time.Second).
 		SetMonitor(apm.NewMonitor(apm.WithCommandAttributeDisabled(false), apm.WithCommandAttributeTransformer(redactSensitiveCollections)))
 
-	if settings.HasAuth() {
-		ymlUser, ymlPwd, err := settings.GetAuth()
+	if settings.AWSAuthEnabled {
```

Comment:
What's the downside to this always being enabled?

Reply:
Ah, I was just about to comment on the PR: I discovered from testing that in CI the tests were able to get AWS creds from the EC2 metadata and then when they tried to use them to connect to the test db the db rejected the connection because it hadn't been configured with AWS auth (e.g. [here](https://parsley.mongodb.com/evergreen/evergreen_ubuntu2204_test_cloud_patch_d46bf5ad9bb68de98cdb491c2d802515a73a7dd3_6568a7b7c9ec44e11507355f_23_11_30_15_18_18/0/task?bookmarks=0,610&shareLine=566)).

The solution I tried to implement is to add a flag that will allow us to only attempt to use AWS auth when it's specifically been configured. This way if the app is running in an environment that has the db configured to use auth we'll explicitly enable it.
The alternative was to let us turn it off for testing, but it made more sense to me to turn it on specifically when we want it.

Reply:
nice! thank you!